grader/lib/cli.py

```diff
@@ -66,6 +67,17 @@ def parse_assignment(args: str, assignments: Set[Assignment]) -> Optional[Assign
 
     error('unknown test: {}'.format(args))
 
+def parse_truncate_range(arg: str) -> int:
+    try:
+        value = int(arg)
+
+        if value < 0:
+            raise ArgumentTypeError("truncate line count must be a natural number")
+
+        return value
+    except ValueError:
+        raise ArgumentTypeError("truncate line count must be an integer")
+
 
 def validate_options_for(assignment: Optional[Assignment]):
     if not bulk_grade_mode and is_in_quiet_mode() and assignment is None:
```

Comment:
It seems like this is the culprit. Interestingly, it only emits an error that no assignment has been specified if the grader is not in bulk-grading mode (that's fine) _and it is in quiet mode_.
Interestingly, function `error` does not even print the message loudly (i.e. ignoring quiet), but rather silently exits.
It looks like the call to `is_in_quiet_mode()` is not even necessary. This is the only position where the quiet mode state is queried and thus entering it does not have any effect on whether the assignment name is required or not.